Source/Settings.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Return the value of a global setting for a given key.
+ * @param key The setting to look up.
+ * @return The setting value.
+ *
+ * This is equivalent to reading a setting for the currently
+ * running version of Sketch using the `defaults` command line tool,
+ * eg: defaults read com.bohemiancoding.sketch3 <key>
+ * */
+export function globalSettingForKey(key) {
+  const value = NSUserDefaults.standardUserDefaults().objectForKey_(key)
+  if (typeof value === 'undefined' || value === 'undefined') {
+    return undefined
+  }
+  return JSON.parse(value)
+}
+
+/**
+ * Set the value of a global setting for a given key.
+ *
+ * @param key The setting to set.
+ * @param value The value to set it to.
+ *
+ * This is equivalent to writing a setting for the currently
+ * running version of Sketch using the `defaults` command line tool,
+ * eg: defaults write com.bohemiancoding.sketch3 <key> <value>
+ */
+export function setGlobalSettingForKey(key, value) {
+  NSUserDefaults.standardUserDefaults().setObject_forKey_(
+    JSON.stringify(value) || 'undefined',
+    key
+  )
+}
+
+const SUITE_PREFIX = 'plugin.sketch.'
+
+/**
+ * Return the value of a global setting for a given key.
+ * @param context The context dictionary passed to the script when it was invoked.
```

Comment:
Are there cases where `context` will change? Otherwise, having to pass the `context` feels a bit pointless, as we already have it baked into `Application`. Would it violate the API design to have `Settings` owned by `Application` and instantiated with `context`?

Reply:
Ah I'm writing a post to explain the philosophy behind those changes, give me a day :)

As a quick answer, you can call those methods with `context.api.Settings.settingForKey` so it's not hoisted in the Application anymore